### PR TITLE
feat: add confirm upgrade modal for flat rate subscriptions

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Problem

We've received many complaints that users did not know they would be billed immediately when they clicked add to the teams adoon. This adds a confirmation modal for flat rate subscription products to confirm the details and tell the user they will be billed immediately.

This will only show for the Teams Addon right now since it's the only flat product we have. 

Screenshot 2024-05-23 at 11 49 31 AM

### CODE REVIEW

File: frontend/src/scenes/billing/BillingProductAddon.tsx [hunk 1]
```diff
@@ -39,6 +41,12 @@ export const BillingProductAddon = ({ addon }: { addon: BillingProductV2AddonTyp
         addon.type === 'enhanced_persons' &&
         addon.plans?.find((plan) => plan.plan_key === 'addon-20240404-og-customers')
 
+    const handleUpgrade = (plan: BillingV2PlanType): void => {
+        window.location.href = `/api/billing-v2/activation?products=${addon.type}:${plan?.plan_key}${
+            redirectPath && `&redirect_path=${redirectPath}`
+        }`
+    }
```
Comment (raquelmsmith) [inline frontend/src/scenes/billing/BillingProductAddon.tsx:48]: this feels like it should be in a logic somewhere

File: frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    <LemonButton type="primary" onClick={() => confirm()}>
+                        Sign me up
+                    </LemonButton>
+                </>
+            }
+        >
+            <div className="max-w-120">
+                <p>
+                    Woo! You're gonna love the {upgradePlan?.name}. We're just confirming that this is a $
+                    {Number(upgradePlan?.unit_amount_usd)}/{upgradePlan?.unit} subscription and the first payment will
```
Comment (raquelmsmith) [inline frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx:30]: The amount they are charged today is actually a prorated rate based on the remaining time in their billing period. So we need to calculate the time remaining in their billing period (if they have an active sub) and then go from there to get the approximate prorated amount.

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
This makes sense now 😄 when it was a plan there was a note about this, but we lost that when it moved to an addon. 

This looks good, just one change 👍

File: frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx (hunk 2 above)
Comment (zlwaterfield) [reply frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx:30]: Do we have that calculation existing any where on the client already?

Comment (raquelmsmith) [reply frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx:30]: I think it used to be in the pricing comparison cable but I took a quick peek there and didn't see it. But it's worth a deeper look.

**Comment by zlwaterfield** [comment]
Updated copy

Screenshot 2024-05-23 at 5 00 33 PM

**Review by xrdt — APPROVED** [review]

**Comment by zlwaterfield** [comment]
Showing all features now:
Screenshot 2024-05-24 at 4 15 58 PM
Screenshot 2024-05-24 at 4 16 01 PM

File: frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { confirmUpgradeModalLogic } from './confirmUpgradeModalLogic'
+
+export function ConfirmUpgradeModal(): JSX.Element {
+    const { upgradePlan } = useValues(confirmUpgradeModalLogic)
+    const { daysRemaining, daysTotal, billing } = useValues(billingLogic)
+    const { hideConfirmUpgradeModal, confirm, cancel } = useActions(confirmUpgradeModalLogic)
+
+    const prorationAmount = useMemo(
+        () =>
+            upgradePlan?.unit_amount_usd
```
[the remark this answers is not in the document]
Comment (zlwaterfield) [reply frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx:16]: Yep I think that's correct.

**Review by raquelmsmith — APPROVED** [review]
I feel like this probably could have been a single line of text instead of a dedicated modal. But 👍 with this if you feel like you really want it to be a modal!

**Comment by zlwaterfield** [comment]
I thought the same but we've had enough complaints that I want this to be very clear. This will also allow us to use it in other spots in the application.

**Pull request merged**